src/Backups/BackupCoordinationStageSync.cpp

```diff
@@ -870,6 +875,9 @@ bool BackupCoordinationStageSync::checkIfHostsReachStage(const Strings & hosts,
             continue;
         }
 
+        if (state.host_with_error)
+            std::rethrow_exception(state.hosts.at(*state.host_with_error).exception);
+
```

Comment:
This change is to fix the [failure](https://s3.amazonaws.com/clickhouse-test-reports/71947/ade2dadd01098c6c3784ff28d4687f6d16ed0b15/integration_tests__tsan__[3_6].html) of test `test_backup_restore_on_cluster/test_cancel_backup.py::test_cancel_restore`.